nixos/modules/misc/ids.nix

```diff
@@ -356,6 +356,7 @@ in
       rstudio-server = 324;
       localtimed = 325;
       automatic-timezoned = 326;
+      vault-agent = 327;
```

Comment:
@pennae What is "dynamic assignment (distinct from DynamicUser!)"?

If you mean just declaring a `users.users.*` without specifying an ID, and letting NixOS create a user with arbitray ID: That is bad practice because each system may then get a different users. That creates all kinds of problems, including that you

* cannot `rsync -a` `/var` data across servers in a fleet
* cannot restore NixOS from a backup of `/var` on the same machine because you might have different IDs this time

Reply:
> What is "dynamic assignment (distinct from DynamicUser!)"?
> 
> If you mean just declaring a `users.users.*` without specifying an ID, and letting NixOS create a user with arbitray ID

Yes, exactly that.

Not sure what to tell you @nh2, but this is policy in nixpkgs.

IIRC this is only really mentioned in https://github.com/NixOS/nixpkgs/blob/a1fd087b05f9061cb75d8e912d9cc879bb24ee2b/nixos/modules/misc/ids.nix#L1-L10 and has no dedicated manual section or anything.